### ISSUE TITLE
Formatting issues (dots replaced by underscores) in namespaces generated by the SPA templates.

### ISSUE THREAD

**Issue by ehouarn-perret** [aspnet/JavaScriptServices#1225]
When creating a new solution and say a project with name like `SolutionName.ProjectName` using one of the SPA templates, the generation of the `*.cs` files (and maybe some others?) ends up with an underscore instead of the original dot in the namespaces, e.g. `SolutionName_ProjectName`. Is there anything that can be done in that regard?

**Comment by SteveSandersonMS** [comment]
Hi there. I've just checked, and the only file that gets its name from your project name is the `csproj` itself (e.g., `MyApplication.csproj`).

Using the templates built into .NET Core 2.0 SDK (both via `dotnet new` and via Visual Studio), I just verified that project names containing `.` characters are *not* mapped to underscores. I'm getting `csproj` files with names like `My.Super.Spa.csproj` exactly as expected.

So, I'm guessing you must be using an older version of the templates, possibly even via Yeoman. Could you make sure you get the latest .NET Core 2.0 SDK and try it with that (*not* with Yeoman - that's deprecated)?

**Comment by SteveSandersonMS** [comment]
Sorry, I re-read your issue and realised you are talking about *namespaces* specifically. Yes, you're right, something does look odd with that. I'll look into it.

**Comment by ehouarn-perret** [comment]
@SteveSandersonMS yes project files are actually okay, just the namespaces are a bit weird. 
Didn't dig much into the generation process though.

**Comment by ehouarn-perret** [comment]
@SteveSandersonMS I may know partially why, tell me if I am wrong but the generator creates also the `package.json`, however, there is a restriction (at least I got a warning cringed by VS 2k17 or R#) about the `name` attribute that can assigned (maybe just informal) that the `.` is not that part of the characters accepted by the Regex (noticed that the dots were also changed into underscores just like the namespaces in the `package.json`).
This is also referenced here: 
 * https://github.com/aspnet/Home/issues/1315
 * https://github.com/aspnet/Templates/issues/369

**Comment by TerryDieckmann** [comment]
@SteveSandersonMS & @ehouarn-perret I am experiencing this issue too, not only with SPA applications, but with my own templates aswell. My `template.json` file looks exactly as described in [this article](https://blogs.msdn.microsoft.com/dotnet/2017/04/02/how-to-create-your-own-templates-for-dotnet-new/). 
<br>The weird thing is, dotnet new works perfectly fine with `dotnet new console -n "Some.Project.With.Dots"`. <br>I didn't test other templates, but I know for sure that my own templates aren't working when I use dots in their names.
